lib/problem_downloader.rb

```diff
@@ -14,7 +20,7 @@ class NoDownloadStrategy < Exception; end
 
   def initialize(oj_name, problem_index, opts={})
     @oj_name = oj_name.downcase.to_s
-    @problem_index = problem_index.to_i
+    @problem_index = problem_index.to_s
```

Comment:
In order to support SPOJ, Codeforces, AIZU and so on, I modify the data type of problem_index to String. You should think about is there anything else should also be modified.